### PR TITLE
[MI-3027] Fix issue 15 of handling lifecycle subscriptionRemoved event

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Fixed issue of handling lifecycle "subscriptionRemoved" event

#### Ticket Link
https://github.com/mattermost/mattermost-plugin-msteams-sync/issues/15

### CODE REVIEW

File: server/api.go [hunk 1]
```diff
@@ -138,8 +138,17 @@ func (a *API) processLifecycle(w http.ResponseWriter, req *http.Request) {
...
+			if err != nil {
+				a.p.API.LogError("Unable to subscribe to channels", "error", err)
+			}
+
+			_, err = a.p.msteamsAppClient.SubscribeToChats(a.p.GetURL()+"/", a.p.configuration.WebhookSecret, !a.p.configuration.EvaluationAPI)
+			if err != nil {
+				a.p.API.LogError("Unable to subscribe to chats", "error", err)
+			}
 		}
-		// TODO: handle "missed" lifecycle event to resync
```
Comment (jespino) [inline server/api.go:142]: we are not handling yet the `missed` event, so I would prefer to keep this comment here to remember that.

**Review by jespino — CHANGES_REQUESTED** [review]
Tiny change, but otherwise looks good to me.

File: server/api.go (hunk 1 above)
Comment (ayusht2810) [reply server/api.go:142]: the resources we are using don't support `missed` lifecycle notification event
you can read more about it here: https://learn.microsoft.com/en-us/graph/webhooks-lifecycle?tabs=http#resources-that-support-lifecycle-notifications

**Comment by ayusht2810** [comment]
@jespino replied to the comment

**Review by jespino — APPROVED** [review]
LGTM

**Pull request merged**